apluslms_roman/cli.py

```diff
@@ -379,6 +413,142 @@ def config_print_action(context):
         print("---\n# project config")
         yaml_dump(get_config(context)._data, stdout)
 
+def init_config(context):
+    cwd = getcwd()
+    # don't create if config already exists
+    try:
+        config_path = RomanConfig.find_from(cwd).path
+        # if the folder is legacy, find_from won't error
+        # so we need to check if the file actually exists
+        if not isfile(config_path):
+            raise RomanConfigError
+        exit(1, "{} already exists in {}.\nNo new Roman config created.".format(
+            basename(config_path), cwd)
+        )
+    except RomanConfigError:
```

Comment:
I would prefer this logic to be in `configuration.py`, so it would probably be easier to use from the GUI too.

That said, I think this should be easy as `RomanConfig.load(join(getcwd(), RomanConfig.DEFAULT_FILENAME, allow_missing=True).save()`. We don't need to hangle legacy_config in this case. The whole code should be something like:

```
try:
  try:
    RomanConfig.find_from(getcwd()).save(create=True) # create=True uses 'x' instead of 'w' for `open`
  except RomanConfigError:
    RomanConfig.create(join(getcwd(), RomanConfig.DEFAULT_FILENAME))
except FileExistsError:
  exit(1, "file exists")
```

And create classmethod in Document:

```
@classmethod
def create(self, filename):
  document = cls.load(filename, allow_missing=True)
  document.save(create=True) # create=True uses 'x' instead of 'w' for `open`
  return document
```

`find_from` should set a real path when it loads legacy configuration in memory, so we can use `save`. In addition, `DEFAULT_PREFIXES`, `DEFAULT_NAMES` and `DEFAULT_FILENAME` should probably be defined in the RomanConfig class, so they are easier to use from outside.

Reply:
I personally would find `overwrite=False` more intuitive, would that work?

Reply:
True, that seems better. I maybe prefer default values to be False or None, but I think this is good place to deviate from that. `create=True` is misleading at least. It that would be used, then it should be `fail_if_exists` or `require_create` and both are a bit unclear, thus `overwrite=True` or `allow_overwrite=True` is probably the best.